lib/hotel.rb

```diff
@@ -0,0 +1,183 @@
+require_relative "room"
+require_relative "reservation"
+require_relative "date_range"
+require_relative "hotel_block"
+require "date"
+
+module HotelSystem
+  class Hotel
+    attr_reader :rooms, :blocks
+    attr_accessor :reservations
+
+    def initialize(number_of_rooms)
+      @rooms = []
+      add_rooms(number_of_rooms)
+      @reservations = []
+      @blocks = []
+    end
+
+    def add_rooms(number)
+      if number.class != Integer || number < 1
+        raise ArgumentError, "Please enter a number greater than 1."
+      end
+      (1..number).each do |num|
+        room = HotelSystem::Room.new(num)
+        @rooms << room
+      end
+    end
+
+    def valid_date_entry?(year, month, day)
+      HotelSystem::DateRange.valid_date_entry?(year, month, day)
+    end
+
+    def create_date_range(start_year:, start_month:, start_day:, num_nights: nil)
+      valid_date_entry?(start_year, start_month, start_day)
+      dates = HotelSystem::DateRange.new(start_year: start_year, start_month: start_month, start_day: start_day, num_nights: num_nights)
+      return dates
+    end
+
+    def find_room(number)
+      HotelSystem::Room.valid_room_number(number)
+      return @rooms.find { |room| room.room_number == number }
+    end
+
+    def room_reserved?(room_number:, year:, month:, day:)
+      valid_date_entry?(year, month, day)
+      res_room = find_room(room_number)
+      res_date = Date.new(year, month, day)
+      if res_room.reservations == []
+        return false
+      end
+      res_room.reservations.each do |res|
```

Comment:
You don't need this explicit check for an empty reservation list. If the array is empty, then the loop starting on line 51 will execute zero times, spitting you out at `return false` on line 56.